linera-core/src/worker.rs

```diff
@@ -613,6 +613,9 @@ where
         let tip = chain.tip_state.get_mut();
         tip.block_hash = Some(certificate.hash());
         tip.next_block_height.try_add_assign_one()?;
+        tip.num_incoming_messages += block.incoming_messages.len() as u32;
+        tip.num_operations += block.operations.len() as u32;
+        tip.num_outgoing_messages += messages.len() as u32;
```

Comment:
I'd add the same operations to `handle_block_proposal`, but without actually assigning them, and with checked operations instead, so that `handle_block_proposal` returns an `ArithmeticError::Overflow` if any of these additions or `u32` conversions would overflow. Something like:

```rust
let num_operations = u32::try_from(block.operations.len()).ok_or(ArithmeticError::Overflow);
tip.num_operations.checked_add(num_operations).ok_or(ArithmeticError::Overflow);
```

Reply:
@afck 
I added the recommended checks in `handle_block_proposal`(`verify_block_chaining`).

Reply:
Thanks! But we don't have the outgoing messages at that point yet; so maybe it would be better to do the checks at the end of `handle_block_proposal` itself?

Reply:
Yes, @afck 
Here is another commit for updating the `handle_block_proposal`.
https://github.com/linera-io/linera-protocol/pull/1397/commits/234c802f192b12f1c3e65543742a22c4bc163d87

Reply:
Thanks!
Sorry, I didn't think this through: Since you added the fields to `ChainTipState`, it _does_ make sense to move the checks into a method of that, just a separate one, not `verify_block_chaining`. Maybe `verify_counters`?
(Not a blocker from my side. Sorry for the confusion!)